code/Control/UserDefinedFormController.php

```diff
@@ -574,43 +576,36 @@ private static function getRestrictedAccessField(string $folder, string $title)
 
     /**
      * This returns a Confirm Folder form used to verify the upload folder for EditableFileFields
-     *
-     * @return ViewableData
+     * @param HTTPRequest $request
+     * @return HTTPResponse
      */
-    public function confirmfolderform()
+    public function confirmfolderformschema(HTTPRequest $request)
     {
-        $request = $this->getRequest();
+        // Retrieve editable form field by its ID
         $id = $request->requestVar('ID');
         if (!$id) {
             throw new HTTPResponse_Exception(_t(__CLASS__.'.INVALID_REQUEST', 'This request was invalid.'), 400);
         }
-        $userFormID = $request->requestVar('UserFormID');
```

Comment:
Wwe no longer care about the UserFormID value. We'll get the parent UserForm by finding the EditableField and fetching its parent.